integration-tests/src/test/java/oracle/kubernetes/operator/ITSessionMigration.java

```diff
@@ -0,0 +1,345 @@
+      httpAttrMap.put("primary", "(.*)primary>(.*)</primary(.*)");
+      httpAttrMap.put("secondary", "(.*)secondary>(.*)</secondary(.*)");
+      httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+
+      // Build WAR in the admin pod and deploy it from the admin pod to a weblogic target
+      domain.buildDeployJavaAppInPod(
+          testAppName, scriptName, BaseTest.getUsername(), BaseTest.getPassword());
+
+      // Wait some time for deployment gets ready
+      Thread.sleep(10 * 1000);
+    }
+  }
+
+  /**
+   * Releases k8s cluster lease, archives result, pv directories
+   *
+   * @throws Exception
+   */
+  @AfterClass
+  public static void staticUnPrepare() throws Exception {
+    if (!QUICKTEST) {
+      logger.info("++++++++++++++++++++++++++++++++++");
+      logger.info("BEGIN");
+      logger.info("Run once, release cluster lease");
+
+      tearDown();
+
+      logger.info("SUCCESS");
+    }
+  }
+
+  /**
+   * Verify that a new and running managed server is picked up as the primary server after the
+   * primary server shut down
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testRepickPrimary() throws Exception {
+    Assume.assumeFalse(QUICKTEST);
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    Map<String, Object> domainMap = domain.getDomainMap();
+    String domainNS = domainMap.get("namespace").toString();
+    String domainUid = domain.getDomainUid();
+
+    String testAppPath = testAppName + "/CounterServlet";
+    String sessCreateTime = "sessioncreatetime";
+    String primaryServ = "primary";
+
+    // Send the first HTTP request and save HTTP header/sessionID info
+    ExecResult result = getHTTPResponse(testAppPath, " -D ");
+
+    // Get primary server name & session create time bf primaryServName1 is stopped
+    String primaryServName1 = getHttpResponseAttribute(result.stdout(), primaryServ);
+    String sessCreateTime1 = getHttpResponseAttribute(result.stdout(), sessCreateTime);
+
+    // Stop primary server
+    stopPrimaryServer(primaryServName1);
+
+    // Send the second HTTP request using HTTP header/sessionID info save before
+    result = getHTTPResponse(testAppPath, " -b ");
+
+    // Get primary server name & session create time af primaryServName1 is stopped
+    String primaryServName2 = getHttpResponseAttribute(result.stdout(), primaryServ);
+    String sessCreateTime2 = getHttpResponseAttribute(result.stdout(), sessCreateTime);
+
+    // Verify that the same session info is used
+    Assume.assumeTrue("HTTP Session should NOT change!", sessCreateTime1.equals(sessCreateTime2));
+
+    // Verify that a new primary server is picked
+    Assume.assumeFalse(
+        "A new primary server should be picked!",
+        primaryServName1.trim().equals(primaryServName2.trim()));
+
+    // Restore test env
+    // Wait some time for ms pod to become ready
+    Thread.sleep(30 * 1000);
```

Comment:
sleep 30 secs is not needed here, TestUtils.checkPodReady checks in loop. This adds up test run time. @hzhao-github @anpanigr

Reply:
agreed. However, if there no any waiting time, the test will start before the pod become ready. Tried 10 seconds but not working.

Reply:
TestUtils.checkPodReady waits upto 250 seconds for the pod to become ready in loop. which test starts before the pod is ready?

Reply:
you are right, TestUtils.checkPodReady waits upto 250 seconds, not 50 seconds as I mentioned below. Let me remove the sleep time here. If the test still fails, we need to file a bug for too long for a pod to become ready online